packages/create-workleap/src/bin.ts

```diff
@@ -1,111 +1,92 @@
 #!/usr/bin/env node
-import * as child_process from "child_process";
-import process from "process";
-import { spinner } from "@clack/prompts";
-
-import { intro, select, note, text, outro } from "./prompts";
-import * as pkg from "../package.json";
-
-const DefaultOutputDirectory = "./my-new-project";
-const NameParameterPosition = 2; // TODO validate position of parameter once ask with `pnpm create`
-
-const FoundryCmd = "foundry";
-
-const askForOutputDirectoryAsync = async (): Promise<string> => {
-    const outputDirectoryFromArgument: string =
-    process.argv[NameParameterPosition];
+import process from "node:process";
+import * as p from "@clack/prompts";
+import fs from "node:fs";
+import path from "node:path";
+import colors from "picocolors";
+import { AvailableTemplates, generateProject, type TemplateId } from "./generateProject.js";
+import packageJson from "../package.json" assert { type: "json" };
+
+// This type is required until this PR has been merged in the @clack/prompts package:
+// https://github.com/natemoo-re/clack/pull/102
+interface Option<Value> {
+    value: Value;
+    label?: string;
+    hint?: string;
+}
+
+let outputDir = process.argv[2];
+
+p.intro(colors.gray(`${packageJson.name} - v${packageJson.version}`));
+
+// Ask for output directory
+if (!outputDir) {
+    const dir = await p.text({
+        message: "Where should we create the project?",
+        placeholder: "./my-new-project",
+        initialValue: "./my-new-project"
+    });
 
-    if (outputDirectoryFromArgument) {
-        note(`${outputDirectoryFromArgument} project setup`);
+    if (p.isCancel(dir)) {
+        p.cancel("Operation cancelled.");
+        process.exit(1);
     }
 
-    return (
-        outputDirectoryFromArgument ??
-    (await text(
-        "Where should we create the project?",
-        DefaultOutputDirectory,
-        DefaultOutputDirectory
-    ))
-    );
-};
-
-const askForTemplateAsync = (): Promise<string> => {
-    const templates = [
-        { value: "host-application" },
-        { value: "remote-module" },
-        { value: "static-module" }
-    ];
+    outputDir = dir as string;
+}
 
-    return select<string>("Select the template to create", templates);
-};
-
-const askForScopeAsync = (template: string): Promise<string> => {
-    return text(
-        `What should be the ${template} scope?`,
-        "Press enter if no scope is needed."
-    );
-};
-
-const callFoundryAsync = async (
-    outputDirectory: string,
-    template: string,
-    scope: string
-): Promise<number> => {
-    const options = [template];
-
-    if (outputDirectory) {
-        options.push("-o", outputDirectory);
-    }
+// Check if the directory is empty
+if (fs.existsSync(outputDir)) {
```

Comment:
If you want, you can use the `async` version of `fs` using `fs/promises`.
But doing that, `fs.exist()` will be replace by `fs.access()` since `exist` (only the async version) is deprecated.